packages/flutter_tools/lib/src/flutter_plugins.dart

```diff
@@ -234,26 +234,79 @@ bool _writeFlutterPluginsList(
     'macos': swiftPackageManagerEnabledMacos,
   };
 
-  // Only notify if the plugins list has changed. [date_created] will always be different,
-  // [version] is not relevant for this check.
-  final String? oldPluginsFileStringContent = _readFileContent(pluginsFile);
-  bool pluginsChanged = true;
-  if (oldPluginsFileStringContent != null) {
-    try {
-      final Object? decodedJson = jsonDecode(oldPluginsFileStringContent);
-      if (decodedJson is Map<String, Object?>) {
-        final String jsonOfNewPluginsMap = jsonEncode(pluginsMap);
-        final String jsonOfOldPluginsMap = jsonEncode(decodedJson[_kFlutterPluginsPluginListKey]);
-        pluginsChanged = jsonOfNewPluginsMap != jsonOfOldPluginsMap;
-      }
-    } on FormatException catch (_) {}
+  // Only write the plugins file if its content have changed. This ensures
+  // that flutter assemble targets that depend on this file aren't invalidated
+  // unnecessarily.
+  final (:bool pluginsChanged, :bool contentsChanged) = _detectFlutterPluginsListChanges(
+    pluginsFile,
+    result,
+  );
+  if (contentsChanged) {
```

Comment:
Nit: I know that `contentsChanged` should always be true if `pluginsChanged` is true, but this reads a little better imo and is a bit more defensive against future updates. I don't feel strongly, though.

```suggestion
  if (pluginsChanged || contentsChanged) {
```

Reply:
Sure, seems reasonable to me!